clang/test/SemaSYCL/num_simd_work_items_device.cpp

```diff
@@ -164,57 +186,42 @@ struct TRIFuncObjBad18 {
 };
 
 #endif // TRIGGER_ERROR
-// If the declaration has an [[intel::reqd_work_group_size]] or
-// [[cl::reqd_work_group_size]] attribute, tests that check if
-// the work group size attribute argument (the first argument)
-// can be evenly divided by the num_simd_work_items attribute.
+// If the declaration has a [[intel::reqd_work_group_size()]]
+// or [[cl::reqd_work_group_size()]] attribute, check to see
+// if the last argument can be evenly divided by the
+// [[intel::num_simd_work_items()]] attribute.
+// If the declaration has a __attribute__((reqd_work_group_size))
+// attribute, check to see if the first argument can be evenly
+// divided by the [[intel::num_simd_work_items()]] attribute.
+// GNU spelling of ReqdWorkGroupSizeAttr maps to the OpenCL semantics.
+// First and last argument are only swapped for the Intel atributes in
+// SYCL and not the OpenCL ones.
 struct TRIFuncObjGood1 {
   [[intel::num_simd_work_items(4)]]
-  [[intel::reqd_work_group_size(64, 64, 5)]] void
+  [[intel::reqd_work_group_size(3, 64, 4)]] void
   operator()() const {}
 };
 
 struct TRIFuncObjGood2 {
-  [[intel::reqd_work_group_size(64, 64, 5)]]
+  [[intel::reqd_work_group_size(3, 64, 4)]]
   [[intel::num_simd_work_items(4)]] void
   operator()() const {}
 };
 
 struct TRIFuncObjGood3 {
   [[intel::num_simd_work_items(4)]]
-  [[cl::reqd_work_group_size(64, 64, 5)]] void
+  [[cl::reqd_work_group_size(3, 64, 4)]] void
   operator()() const {}
 };
 
 struct TRIFuncObjGood4 {
-  [[cl::reqd_work_group_size(64, 64, 5)]]
+  [[cl::reqd_work_group_size(3, 64, 4)]]
   [[intel::num_simd_work_items(4)]] void
   operator()() const {}
 };
 
-struct TRIFuncObjGood5 {
```

Comment:
Why remove this test (I think it tests something interesting)?

Reply:
> Why remove this test (I think it tests something interesting)?

@AaronBallman  I removed them because they are failing now, so added them as BAD tests. We are giving optional values here:

```
 //If no attribute argument is specified, set the second and third argument
 // to the default value 1, but only if the intel::reqd_work_group_size spelling was used.

struct TRIFuncObjGood5 {
  [[intel::num_simd_work_items(4)]]
  [[intel::reqd_work_group_size(64)]]  //identical to  [[intel::reqd_work_group_size(64, ,1, 1)]]
  void operator()() const {}
};
struct TRIFuncObjGood7 {
  [[intel::num_simd_work_items(4)]]
  [[intel::reqd_work_group_size(5, 64)]] //  //identical to  [[intel::reqd_work_group_size(5, ,64, 1)]]  - we should not diagnosis 
  void operator()() const {}
};
```

Reply:
Ah, I see that you moved them -- I added comments on the new tests. What threw me is that these are still good (but we have a different bug to fix before they'll start to pass again).

Reply:
> Ah, I see that you moved them -- I added comments on the new tests. What threw me is that these are still good (but we have a different bug to fix before they'll start to pass again).

Yes, that's what i am thinking too. This is creating another bug here.